bundles/AdminBundle/Controller/Admin/NotificationController.php

```diff
@@ -0,0 +1,222 @@
+
+        return $this->adminJson([
+            'success' => true,
+            'data'    => $data,
+        ]);
+    }
+
+    /**
+     * @param Request $request
+     * @param NotificationService $service
+     * @return JsonResponse
+     * @Route("/find-all")
+     */
+    public function findAllAction(Request $request, NotificationService $service): JsonResponse
+    {
+        $this->checkPermission('notifications');
+
+        $filter = ['recipient = ?' => (int) $this->getAdminUser()->getId()];
+        $parser = new NotificationServiceFilterParser($request);
+
+        foreach ($parser->parse() as $key => $val) {
+            $filter[$key] = $val;
+        }
+
+        $options = [
+            'offset' => $request->get('start', 0),
+            'limit' => $request->get('limit', 40)
+        ];
+
+        $result = $service->findAll($filter, $options);
+
+        $data = [];
+
+        foreach ($result['data'] as $notification) {
+            $data[] = $service->format($notification);
+        }
+
+        return $this->adminJson([
+            'success' => true,
+            'total'   => $result['total'],
+            'data'    => $data,
+        ]);
+    }
+
+    /**
+     * @param Request $request
+     * @param NotificationService $service
+     * @return JsonResponse
+     * @Route("/find-last-unread")
+     */
+    public function findLastUnreadAction(Request $request, NotificationService $service) : JsonResponse
+    {
+        $this->checkPermission('notifications');
+
+        $user     = $this->getAdminUser();
+        $interval = (int) $request->get('interval', 10);
+        $result   = $service->findLastUnread((int) $user->getId(), $interval);
+        $unread   = $service->countAllUnread((int) $user->getId());
+
+        $data = [];
+
+        foreach ($result['data'] as $notification) {
+            $data[] = $service->format($notification);
+        }
+
+        return $this->adminJson([
+            'success' => true,
+            'total'   => $result['total'],
+            'data'    => $data,
+            'unread'  => $unread,
+        ]);
+    }
+    /**
+     * @param Request $request
+     * @param NotificationService $service
+     * @return JsonResponse
+     * @Route("/mark-as-read")
+     */
+    public function markAsReadAction(Request $request, NotificationService $service): JsonResponse
+    {
+        $this->checkPermission('notifications');
+
+        $id = (int) $request->get('id', 0);
+        $service->findAndMarkAsRead($id);
+
+        return $this->adminJson(['success' => true]);
+    }
+    /**
+     * @param Request $request
+     * @param NotificationService $service
+     * @return JsonResponse
+     * @Route("/delete")
+     */
+    public function deleteAction(Request $request, NotificationService $service): JsonResponse
+    {
+        $this->checkPermission('notifications');
+
+        $id = (int) $request->get('id', 0);
+        $service->delete($id);
```

Comment:
make sure that one sees only notifications that are assigned to him and cannot delete notifications from another user.